lib/date-string.js

```diff
@@ -53,6 +53,13 @@ function DateString(value) {
     return new DateString(value);
   }
 
+  if (value instanceof DateString) {
+    return Object.assign(
+      new DateString(value.when),
+      value
+    );
```

Comment:
I find this implementation rather complex and difficult to understand, I am worried about introducing subtle bugs. Have you considered a simpler implementation along the following lines?

```ts
function DateString(value) {
  if (!(this instanceof DateString)) {
    return new DateString(value);
  }

  if (value instanceof DateString) {
    value = value.when;
  }

  if (typeof(value) !== 'string') {
    throw new Error('Input must be a string');
  }

  // etc.
}
```